src/nats.rs

```diff
@@ -152,6 +152,107 @@ pub async fn ensure_stream(
         .map_err(|e| anyhow::anyhow!("{e:?}"))
 }
 
+pub async fn ensure_limits_stream(
+    context: &Context,
+    name: String,
+    subjects: Vec<String>,
+    description: Option<String>,
+) -> Result<Stream> {
+    let stream_config = StreamConfig {
+        name: name.clone(),
+        description,
+        num_replicas: 1,
+        retention: async_nats::jetstream::stream::RetentionPolicy::Limits,
+        subjects,
+        max_age: DEFAULT_EXPIRY_TIME,
+        storage: async_nats::jetstream::stream::StorageType::File,
```

Comment:
Nit: Would memory be a better default since it isn't a huge deal if the events drop? It serves more as a collector anyway

Reply:
I haven't spend any time looking at the implications between the two storage options, but I'm fine with switching this to Memory until we have evidence pointing otherwise 👍

Reply:
After talking with @joonas offline, turns out there are tradeoffs and configurations needed to use memory storage types. So we're not going to worry about it for now